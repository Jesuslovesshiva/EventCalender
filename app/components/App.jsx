import {Outlet, ReactLocation, Router} from '@tanstack/react-location'
import React from 'react'

import ToastContainer from '@/components/atoms/ToastContainer'
import ReloadPrompt from '@/components/organisms/ReloadPrompt'
import {history} from '@/configure-store'
import EntitiesInjector from '@/injectors/entities/EntitiesInjector'

import Theme from './atoms/Theme'
import routes from './routes'

const location = new ReactLocation({history})

const App = () => (
  <>
    <Theme />
    <ToastContainer />
    <EntitiesInjector />
    <Router routes={routes} location={location}>
      <Outlet />
      <ReloadPrompt />
    </Router>
  </>
)

export default React.memo(App)
