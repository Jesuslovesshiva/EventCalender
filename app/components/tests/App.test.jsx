import configureTestStore from '@/utils/test-utils/configure-test-store'
import render from '@/utils/test-utils/custom-render'

import App from '../App'

jest.mock('@/components/atoms/Theme', () => () => <div>Theme</div>)
jest.mock('@/components/pages/HomePage/Loadable', () => () => (
  <div>Home Page</div>
))
jest.mock('@/components/organisms/ReloadPrompt', () => () => (
  <div>Reload Prompt</div>
))
jest.mock('@faker-js/faker', () => {
  const originalModule = jest.requireActual('@faker-js/faker')
  return {
    ...originalModule,
    faker: {
      ...originalModule.faker,
      datatype: {
        number: jest.fn(() => 8)
      }
    }
  }
})

const {options} = configureTestStore()
const renderComponent = () => render(<App />, options)

describe('App', () => {
  it('should match the snapshot', async () => {
    const {container, findByText} = renderComponent()
    await findByText('Home Page')
    expect(container).toMatchSnapshot()
  })
})
