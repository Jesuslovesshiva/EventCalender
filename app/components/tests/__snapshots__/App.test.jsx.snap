// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <div>
    Theme
  </div>
  <div
    class="Toastify"
  />
  <div
    class="flex flex-col h-full bg-background overflow-x-auto"
  >
    <div
      class="bg-white shadow flex justify-between z-50"
    >
      <a
        class="w-[5.625rem] flex items-center justify-center min-w-[50px] active"
        href="/"
      >
        <img
          alt="Logo"
          class="h-8"
          src="SVG_PATH"
        />
      </a>
      <div
        class="flex-1 text-white relative"
      >
        <div
          class=" absolute top-0 bottom-0 right-[100%] w-[8px] pointer-events-none overflow-hidden before:absolute before:inset-0 before:translate-x-[8px] before:shadow-[0_0_8px_0_rgba(0,0,0,0.07),0_1px_1px_0_rgba(0,0,0,0.14)] "
        />
        <nav
          class="flex justify-center flex-1 flex-col lg:flex-row z-10 pt-1 bg-gradient-to-l bg-gradient-primary"
        >
          <div
            class="flex-1 mx-[26px] mt-0.5 w-full order-1"
          >
            <div
              class="text-gray-dark flex items-center justify-start gap-3"
            >
              <div
                class="text-base w-full specific:w-fit css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="!min-h-0 !rounded-1 !transition-shadow pl-3 gap-1.5 !bg-transparent !border-gray-medium hover:!border-surface hover:!bg-silver hover:!shadow-md !rounded-full !border-white !bg-transparent h-[27px] hover:!border-white hover:!bg-transparent hover:!shadow-md css-1s2u09g-control"
                >
                  <div
                    class="!py-0 !px-0 css-319lph-ValueContainer"
                  >
                    <div
                      class="!m-0 !p-0 !text-gray-dark !text-white !mb-px css-qc6sy-singleValue"
                    >
                      U16
                    </div>
                    <div
                      class="!m-0 !p-0 !text-gray-dark !text-white !m-0 !p-0 css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-2-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <div
                      aria-hidden="true"
                      class="text-2xl !px-3 !py-1 !text-gray-medium !text-white !pr-[7px] !pl-0 !py-0 css-tlfecz-indicatorContainer"
                    >
                      <div
                        class="fill-current w-[1em] h-[1em] inline-block"
                        data-icon="arrow_drop_down"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer text-white"
                data-icon="search"
              />
            </div>
          </div>
          <div
            class="flex justify-center overflow-hidden order-last lg:order-2"
          >
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/training-plan"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                Trainingsplan
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/training"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                Training
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/games"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                Games
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/medicine"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                Medicine
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/diagnostics"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                Diagnostics
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/team"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                Team
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/philosophy"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                Philosophy
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/asa"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                ASA
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
            <a
              class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
              href="/administration"
              tabindex="0"
            >
              <div
                class="flex items-center"
              >
                Administration
              </div>
              <div
                class="relative"
              >
                <div
                  class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
                />
              </div>
            </a>
          </div>
          <div
            class=" flex justify-start lg:justify-end items-center justify-self-start flex-1 gap-3 mt-0.5 mb-1.5 mx-[26px] order-2 lg:order-3 "
          >
            <div
              class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
              data-icon="dns"
            />
            <div
              class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
              data-icon="perm_contact_calendar"
            />
            <div
              class="relative"
            >
              <div
                class="fill-current w-[1em] h-[1em] inline-block text-2xl cursor-pointer"
                data-icon="notifications"
              />
              <div
                class=" bg-silver shadow h-3 w-3 text-primary text-xs flex items-center justify-center rounded-full absolute top-0 right-0 -mr-1 -mt-1 "
              >
                8
              </div>
            </div>
            <div
              class=" h-7 w-7 pb-px font-heading-small-caps flex justify-center items-center border rounded-full border-white select-none hover:cursor-pointer "
            >
              JD
            </div>
          </div>
        </nav>
      </div>
    </div>
    <div
      class="flex-1 overflow-y-auto"
    >
      <div>
        Home Page
      </div>
    </div>
  </div>
  <div>
    Reload Prompt
  </div>
</div>
`;
